[18.0][MIG] pos_customer_required: Migration to 18.0

P-H-Phuc opened the pull request:
This module was written to extend the functionality of odoo pos and allows you to require a customer for each pos order. In the pos session configuration, you can choose to require the customer for pos orders.

If a customer is not selected, the pos ui will display an message.
In the backend the customer field is required when needed.

Two new options are available:

- Customer 'Required before starting the order';
- Customer 'Required before paying';

## 'Required before starting the order' Option

In the frontend PoS, the default screen is the screen to select customers.

- Users are not allowed to start selling before having selected a customer;

<img width="1908" height="886" alt="pos_customer_required_before_order" src="https://github.com/user-attachments/assets/9109b07f-2c8f-4c30-9b2d-a5c984f95e77" />

## 'Required before paying' Option

In the frontend PoS, the user can start selling, but if the user tries to make payment and if a customer is not selected, the pos ui will display an message.

<img width="1908" height="886" alt="pos_customer_required_before_paying" src="https://github.com/user-attachments/assets/1e129430-ea94-48e2-8f7d-64a6174469c7" />

legalsylvain committed:
[9.0] [PORT + IMP] pos_require_customer (#74)

oca-transbot committed:
OCA Transbot updated translations from Transifex

legalsylvain committed:
new module pos_margin

etobella committed:
[ADD] PoS session pay invoice module

JordiBForgeFlow committed:
Migration of pos_session_pay_invoice to 11.0

PierrickBrun committed:
[ADD] backport session summary report with fr translation

bishalgit committed:
Fix account-payment's github repo url

Commit:
[MIG] pos_margin:Migration to 11.0

Commit:
[IMP] Add pos margin in pos order report

Commit:
[12.0][MIG] pos_session_pay_invoice

legalsylvain committed:
[MIG] pos_margin: Migration to 12.0

meodoo committed:
[MIG] pos_margin: Migration to 13.0

wpichler committed:
[MIG] Migrated to Odoo v 12

pedrocs-exo committed:
Added translation using Weblate (Portuguese)

mtbochaca committed:
Added translation using Weblate (Spanish)

mtbochaca committed:
Added translation using Weblate (Catalan)

robyf70 committed:
[NEW][10.0] POS order remove line

robyf70 committed:
[MIG][12.0] POS order remove line

CarlesAntoli committed:
Added translation using Weblate (Catalan)

Commit:
[MIG] pos_customer_required: Migration to 12.0

ivantodorovich committed:
[FIX] Tests and use related field instead of computed

Commit:
Update website

oca-travis committed:
[UPD] Update pos_customer_required.pot

OCA-git-bot committed:
[UPD] README.rst

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: pos-12.0/pos-12.0-pos_customer_required
Translate-URL: https://translation.odoo-community.org/projects/pos-12-0/pos-12-0-pos_customer_required/

CarlesAntoli committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (14 of 14 strings)

Translation: pos-12.0/pos-12.0-pos_customer_required
Translate-URL: https://translation.odoo-community.org/projects/pos-12-0/pos-12-0-pos_customer_required/ca/

CarlesAntoli committed:
Added translation using Weblate (Spanish)

CarlesAntoli committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (12 of 12 strings)

Translation: pos-12.0/pos-12.0-pos_customer_required
Translate-URL: https://translation.odoo-community.org/projects/pos-12-0/pos-12-0-pos_customer_required/es/

joanqubiq committed:
[IMP] pos_order_remove_line: black, isort, prettier

krunal267 committed:
[FIX] Fixed @api.constrains issue

OCA-git-bot committed:
pos_customer_required 12.0.1.0.1

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: pos-12.0/pos-12.0-pos_customer_required
Translate-URL: https://translation.odoo-community.org/projects/pos-12-0/pos-12-0-pos_customer_required/

danimv5 committed:
Added translation using Weblate (Spanish)

jguenat committed:
Added translation using Weblate (French)

acsonefho committed:
[IMP] pos_session_pay_invoice: black, isort, prettier

acsonefho committed:
[13.0][MIG] pos_session_pay_invoice

ivantodorovich committed:
[FIX] Use scss stylesheet instead of less

dsolanki-initos committed:
[IMP] pos_margin: isort, black, prettier.

dsolanki-initos committed:
[MIG] Migrate module pos_margin to v14.

legalsylvain committed:
[IMP] pos_margin : display margin and margin rate in front-end UI

Co-authored-by: Iván Todorovich <ivan.todorovich@gmail.com>

olgamarcocb committed:
[MIG] pos_session_pay_invoice: Ending migration to 13.0

fshah-initos committed:
[IMP] pos_order_remove_line: isort, black , prettier.

fshah-initos committed:
[MIG] Migrate module pos_order_remove_line to v14.

sbidoul committed:
pos_margin: bump version to workround PyPI glitch

Commit:
[IMP] pos_report_session_summary: black, isort, prettier

Commit:
[MIG] pos_report_session_summary: Migration to 14.0

Commit:
[FIX] pos_report_session_summary

olgamarcocb committed:
[IMP] pos_session_pay_invoice: black, isort, prettier

Commit:
[FIX] pos_report_session_summary: do not crash if statement data is missing

olgamarcocb committed:
[MIG] pos_session_pay_invoice: Migration to 14.0

KNVx committed:
[IMP] pos_customer_required: black, isort, prettier

KNVx committed:
[MIG] pos_customer_required: Migration to 14.0

oca-travis committed:
[UPD] Update pos_session_pay_invoice.pot

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 37.5% (3 of 8 strings)

Translation: pos-14.0/pos-14.0-pos_margin
Translate-URL: https://translation.odoo-community.org/projects/pos-14-0/pos-14-0-pos_margin/it/

cvinh committed:
[FIX] Hide margin in Front End if config is false

Commit:
[UPD] Update pos_customer_required.pot

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 37.5% (3 of 8 strings)

Translation: pos-14.0/pos-14.0-pos_margin
Translate-URL: https://translation.odoo-community.org/projects/pos-14-0/pos-14-0-pos_margin/it/

suker committed:
[IMP] : pre-commit stuff

suker committed:
[MIG] pos_margin: Migration to 16.0

suker committed:
[MIG] pos_order_remove_line: Migration to 16.0

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (15 of 15 strings)

Translation: pos-14.0/pos-14.0-pos_customer_required
Translate-URL: https://translation.odoo-community.org/projects/pos-14-0/pos-14-0-pos_customer_required/it/

Ricardoalso committed:
[MIG] pos_report_session_summary: Migration to 15.0

mymage committed:
Added translation using Weblate (Italian)

Commit:
[14.0][IMP] pos_session_pay_invoice: Set `target=new` on views

Commit:
[FIX] pos_session_pay_invoice: `search_paid_order_ids` must exclude orders createds with this module

Commit:
[FIX] `search_paid_order_ids` must exclude orders createds with this module

Commit:
[UPD] Update pos_session_pay_invoice.pot

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (30 of 30 strings)

Translation: pos-14.0/pos-14.0-pos_session_pay_invoice
Translate-URL: https://translation.odoo-community.org/projects/pos-14-0/pos-14-0-pos_session_pay_invoice/it/

OCA-git-bot committed:
pos_session_pay_invoice 14.0.1.1.0

suker committed:
[FIX] pos_order_remove_line: clean refunded orderline of toRefundLines when deleting an orderline from remove-line-button.

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (8 of 8 strings)

Translation: pos-16.0/pos-16.0-pos_margin
Translate-URL: https://translation.odoo-community.org/projects/pos-16-0/pos-16-0-pos_margin/it/

anmarmo1 committed:
Translated using Weblate (Spanish)

Currently translated at 86.6% (26 of 30 strings)

Translation: pos-14.0/pos-14.0-pos_session_pay_invoice
Translate-URL: https://translation.odoo-community.org/projects/pos-14-0/pos-14-0-pos_session_pay_invoice/es/

Commit:
[16.0][FIX] pos_order_remove_line: Trigger

- Function logic is modified in order to use the numpad backspace trigger.
- Stops event propagation and prevents default actions.

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: pos-15.0/pos-15.0-pos_report_session_summary
Translate-URL: https://translation.odoo-community.org/projects/pos-15-0/pos-15-0-pos_report_session_summary/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: pos-14.0/pos-14.0-pos_customer_required
Translate-URL: https://translation.odoo-community.org/projects/pos-14-0/pos-14-0-pos_customer_required/

chienandalu committed:
[FIX] *: Linter fixes

almumu committed:
Is not pIs not possible to print the report. This pr fixes the error: 'str' object has no attribute 'decimal_places' when printing session summaryossible to print the report. This pr fixes the error: 'str' object has no attribute 'decimal_places' when printing session summary

OCA-git-bot committed:
[BOT] post-merge updates

legalsylvain committed:
[FIX] eslint error 'no-shadow' in all modules

legalsylvain committed:
[FIX] eslint error 'no-shadow' in all modules

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (15 of 15 strings)

Translation: pos-14.0/pos-14.0-pos_customer_required
Translate-URL: https://translation.odoo-community.org/projects/pos-14-0/pos-14-0-pos_customer_required/es/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (30 of 30 strings)

Translation: pos-14.0/pos-14.0-pos_session_pay_invoice
Translate-URL: https://translation.odoo-community.org/projects/pos-14-0/pos-14-0-pos_session_pay_invoice/es/

carlos-lopez-tecnativa committed:
[MIG] pos_session_pay_invoice: Migration to 16.0

Commit:
[IMP] pos_report_session_summary: black, isort, prettier

Commit:
[MIG] pos_report_session_summary: Migration to 16.0

Commit:
[UPD] Update pos_report_session_summary.pot

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[UPD] Update pos_session_pay_invoice.pot

OCA-git-bot committed:
[BOT] post-merge updates

BhaveshHeliconia committed:
[IMP] pos_order_remove_line: pre-commit auto fixes

BhaveshHeliconia committed:
[MIG] pos_order_remove_line: Migration to 18.0

antoniodavid committed:
[IMP] pos_margin: pre-commit execution

antoniodavid committed:
[MIG] pos_margin: Migration to 17.0

P-H-Phuc committed:
[IMP] pos_report_session_summary: pre-commit auto fixes

P-H-Phuc committed:
[MIG] pos_report_session_summary: Migration to 18.0

P-H-Phuc committed:
[IMP] pos_customer_required: pre-commit auto fixes

P-H-Phuc committed:
[MIG] pos_customer_required: Migration to 18.0

thienvh332 commented on pos_customer_required/views/pos_order_view.xml:
```diff
@@ -3,7 +3,7 @@
      License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html) -->
 <odoo>
-    <record model="ir.ui.view" id="view_pos_order_form">
+    <record model="ir.ui.view" id="view_pos_pos_form">
```
Do not change the name during migration unless absolutely necessary.

thienvh332 commented on pos_customer_required/views/pos_config_view.xml:
```diff
@@ -3,30 +3,15 @@
      License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html) -->
 <odoo>
-    <record model="ir.ui.view" id="view_pos_config_form">
-        <field name="name">view.pos.config.form</field>
```
Why delete `<field name="name">view.pos.config.form</field>` ?

P-H-Phuc replied:
My mistake

thienvh332 reviewed (commented):
Just some nitpicking :wink:

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (30 of 30 strings)

Translation: pos-16.0/pos-16.0-pos_session_pay_invoice
Translate-URL: https://translation.odoo-community.org/projects/pos-16-0/pos-16-0-pos_session_pay_invoice/it/

Commit:
[UPD] Update pos_session_pay_invoice.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: pos-16.0/pos-16.0-pos_session_pay_invoice
Translate-URL: https://translation.odoo-community.org/projects/pos-16-0/pos-16-0-pos_session_pay_invoice/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (8 of 8 strings)

Translation: pos-16.0/pos-16.0-pos_session_pay_invoice
Translate-URL: https://translation.odoo-community.org/projects/pos-16-0/pos-16-0-pos_session_pay_invoice/it/

carlos-lopez-tecnativa committed:
[IMP] pos_session_pay_invoice: Restrict to journals of the current POS session
To prevent mistakes, users must select only the journals available in the pos.payment.method of the POS session.

Commit:
[UPD] Update pos_session_pay_invoice.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: pos-16.0/pos-16.0-pos_session_pay_invoice
Translate-URL: https://translation.odoo-community.org/projects/pos-16-0/pos-16-0-pos_session_pay_invoice/

carlos-lopez-tecnativa committed:
[IMP] pos_session_pay_invoice: Generate pos.order and pos.payment for customer transactions

Before this commit, the wizard generated an account.bank.statement.line, but the problem was that when the POS had cash control enabled, all transactions generated by this wizard were displayed under Cash, because Odoo considered such statements as Cash In/Out operations, see https://github.com/odoo/odoo/blob/c1d3153ed564c0f47a201e5751740857a15d7db2/addons/point_of_sale/models/pos_session.py#L557-L567.

Now, the transactions are shown under Cash or Bank depending on the payment method and journal type.

For payments of vendor bills, a statement line will still be created, because the POS only works with receivable accounts and cannot reconcile with payable accounts.
See https://github.com/odoo/odoo/blob/c1d3153ed564c0f47a201e5751740857a15d7db2/addons/point_of_sale/models/pos_order.py#L890-L898

Commit:
[UPD] Update pos_session_pay_invoice.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: pos-16.0/pos-16.0-pos_session_pay_invoice
Translate-URL: https://translation.odoo-community.org/projects/pos-16-0/pos-16-0-pos_session_pay_invoice/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (12 of 12 strings)

Translation: pos-16.0/pos-16.0-pos_session_pay_invoice
Translate-URL: https://translation.odoo-community.org/projects/pos-16-0/pos-16-0-pos_session_pay_invoice/it/

P-H-Phuc committed:
[IMP] pos_order_remove_line: Add config show button remove order line

Tisho99 committed:
[IMP] pos_session_pay_invoice: pre-commit auto fixes

Tisho99 committed:
[MIG] pos_session_pay_invoice: Migration to 18.0

Tisho99 committed:
[IMP] pos_margin: pre-commit auto fixes

Tisho99 committed:
[MIG] pos_margin: Migration to 18.0

BernatObrador committed:
[ADD] pos_user_restrict_provider_info: New module pos_user_restrict_provider_info

BernatObrador committed:
[ADD] pos_blind_session_closing: New module pos_blind_session_closing

OCA-git-bot committed:
Merge PR #1472 into 18.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update pos_session_pay_invoice.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #1296 into 18.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update pos_order_remove_line.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #1411 into 18.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update pos_report_session_summary.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (11 of 11 strings)

Translation: pos-18.0/pos-18.0-pos_order_remove_line
Translate-URL: https://translation.odoo-community.org/projects/pos-18-0/pos-18-0-pos_order_remove_line/it/

OCA-git-bot committed:
Merge PR #1495 into 18.0

Signed-off-by ivantodorovich

Commit:
[UPD] Update pos_blind_session_closing.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #1483 into 18.0

Signed-off-by ivantodorovich

Commit:
[UPD] Update pos_margin.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (1 of 1 strings)

Translation: pos-18.0/pos-18.0-pos_blind_session_closing
Translate-URL: https://translation.odoo-community.org/projects/pos-18-0/pos-18-0-pos_blind_session_closing/it/

OCA-git-bot committed:
Merge PR #1490 into 18.0

Signed-off-by ivantodorovich

Commit:
[UPD] Update pos_user_restrict_provider_info.pot

OCA-git-bot committed:
[BOT] post-merge updates
